src/System.Text.Json/tests/PerfSmokeTests.cs

```diff
@@ -0,0 +1,261 @@
+using Xunit;
+
+[assembly: CollectionBehavior(DisableTestParallelization = true)]
+
+namespace System.Text.Json.Tests
+{
+    public class PerfSmokeTests
+    {
+        private static readonly Stopwatch Timer = new Stopwatch();
+        private const bool Collect = true;
+        private const int MemoryToleranceFactor = 5;
+
+        // Keep < 10. Otherwise tests take too long to execute (over 10 seconds).
+        private const int NumberOfIterations = 5;
+
+        private const int NumberOfSamples = 3;
+        private const int ExpectedMemoryBenchMark = 3000000;
+
+        private static readonly bool OutputResults = false;
+
+        [Fact]
+        public void ReadBasicJson()
+        {
+            Output("====== TEST ReadBasicJson ======");
+            RunTest(TestJson.BasicJson, 100 + NumberOfIterations, ExpectedMemoryBenchMark*MemoryToleranceFactor);
+        }
+
+        [Fact]
+        public void ReadProjectLockJson()
+        {
+            Output("====== TEST ReadProjectLockJson ======");
+            RunTest(TestJson.ProjectLockJson, NumberOfIterations*1000, ExpectedMemoryBenchMark*MemoryToleranceFactor);
+        }
+
+        [Fact]
+        public void ReadHeavyNestedJson()
+        {
+            Output("====== TEST ReadHeavyNestedJson ======");
+            RunTest(TestJson.HeavyNestedJson, NumberOfIterations*15, ExpectedMemoryBenchMark*MemoryToleranceFactor);
+        }
+
+        private static void RunTest(string jsonStr, int timeBenchmark, int memoryBenchmark)
+        {
+            var timeIterReadResults = new List<long>();
+            var memoryIterReadResults = new List<long>();
+
+            for (var j = 0; j < NumberOfSamples; j++)
+            {
+                Timer.Restart();
+                for (var i = 0; i < NumberOfIterations; i++)
+                {
+                    // ReSharper disable once UnusedVariable
+                    var json = ReadJson(jsonStr);
+                }
+                var time = Timer.ElapsedMilliseconds;
+                var memory = GC.GetTotalMemory(Collect);
+                timeIterReadResults.Add(time);
+                memoryIterReadResults.Add(memory);
+                Assert.True(time < timeBenchmark);
+                Assert.True(memory < memoryBenchmark);
+            }
+            Output(timeIterReadResults.Average().ToString(CultureInfo.InvariantCulture));
+            Output((memoryIterReadResults.Average()/1000).ToString(CultureInfo.InvariantCulture));
+        }
+
+        private static void Output(string str)
+        {
+            if (!OutputResults) return;
+            Console.WriteLine(str);
+        }
+
+        private static Json ReadJson(string jsonString)
+        {
+            var json = new Json();
+            if (string.IsNullOrEmpty(jsonString))
+            {
+                return json;
+            }
+
+            var jsonReader = new JsonReader(jsonString);
+            var jsonObjectMain = new Object();
```

Comment:
Why are you doing all these allocations? (object, members, etc). I would imagine for a micro-benchmark we would want to simply read the JSON data and verify that what we read is expected. I don't think you want to actually create deserialized objects.